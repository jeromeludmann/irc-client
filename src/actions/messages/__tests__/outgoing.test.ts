import {
  sendPart,
  sendPingToServer,
  sendPongToServer,
  sendQuit,
  sendUser,
  sendJoin,
  sendNick,
  sendPrivmsg,
} from '@app/actions/messages/outgoing'

describe('outgoing messages actions', () => {
  it('should send JOIN', () => {
    expect(sendJoin('serverKey', '#channel')).toMatchSnapshot()
  })

  it('should send NICK', () => {
    expect(sendNick('serverKey', 'new_nick')).toMatchSnapshot()
  })

  it('should send PART', () => {
    expect(sendPart('serverKey', '#channel')).toMatchSnapshot()
  })

  it('should send PART with message', () => {
    expect(sendPart('serverKey', '#channel', 'Goodbye!')).toMatchSnapshot()
  })

  it('should send PING', () => {
    expect(sendPingToServer('serverKey', 'key')).toMatchSnapshot()
  })

  it('should send PONG', () => {
    expect(sendPongToServer('serverKey', 'key')).toMatchSnapshot()
  })

  it('should send PRIVMSG', () => {
    expect(
      sendPrivmsg('serverKey', '#channel', 'hello world'),
    ).toMatchSnapshot()
  })

  it('should send QUIT', () => {
    expect(sendQuit('serverKey')).toMatchSnapshot()
  })

  it('should send QUIT with message', () => {
    expect(sendQuit('serverKey', 'Goodbye!')).toMatchSnapshot()
  })

  it('should send USER', () => {
    expect(sendUser('serverKey', 'username', 'Real Name')).toMatchSnapshot()
  })
})
