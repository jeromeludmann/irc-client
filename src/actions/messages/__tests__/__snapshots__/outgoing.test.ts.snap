// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outgoing messages actions should send JOIN 1`] = `
Object {
  "payload": Object {
    "raw": "JOIN #channel",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "serverKey",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`outgoing messages actions should send NICK 1`] = `
Object {
  "payload": Object {
    "raw": "NICK new_nick",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "serverKey",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`outgoing messages actions should send PART 1`] = `
Object {
  "payload": Object {
    "raw": "PART #channel",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "serverKey",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`outgoing messages actions should send PART with message 1`] = `
Object {
  "payload": Object {
    "raw": "PART #channel Goodbye!",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "serverKey",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`outgoing messages actions should send PING 1`] = `
Object {
  "embeddedAction": Object {
    "payload": Object {
      "key": "key",
    },
    "route": Object {
      "bufferKey": "@status",
      "serverKey": "serverKey",
    },
    "type": "MESSAGE/SEND_PING_TO_SERVER",
  },
  "payload": Object {
    "raw": "PING key",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "serverKey",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`outgoing messages actions should send PONG 1`] = `
Object {
  "embeddedAction": Object {
    "payload": Object {
      "key": "key",
    },
    "route": Object {
      "bufferKey": "@status",
      "serverKey": "serverKey",
    },
    "type": "MESSAGE/SEND_PONG_TO_SERVER",
  },
  "payload": Object {
    "raw": "PONG key",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "serverKey",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`outgoing messages actions should send PRIVMSG 1`] = `
Object {
  "embeddedAction": Object {
    "payload": Object {
      "text": "hello world",
    },
    "route": Object {
      "bufferKey": "#channel",
      "serverKey": "serverKey",
    },
    "type": "MESSAGE/SEND_PRIVMSG",
  },
  "payload": Object {
    "raw": "PRIVMSG #channel :hello world",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "serverKey",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`outgoing messages actions should send QUIT 1`] = `
Object {
  "payload": Object {
    "raw": "QUIT",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "serverKey",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`outgoing messages actions should send QUIT with message 1`] = `
Object {
  "payload": Object {
    "raw": "QUIT Goodbye!",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "serverKey",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;

exports[`outgoing messages actions should send USER 1`] = `
Object {
  "payload": Object {
    "raw": "USER username 0 * :Real Name",
  },
  "route": Object {
    "bufferKey": "@none",
    "serverKey": "serverKey",
  },
  "type": "SOCKET/SEND_RAW_MESSAGE",
}
`;
